### ISSUE TITLE
Spline Smoothing Parameters

### ISSUE THREAD

**Issue by EmmaLiTsai** [EmmaLiTsai/recoverKBTDR#17]
After exploring some smoothing options, I believe that spline smoothing is the best approach for smoothing the trace data. At the moment, the user would visually determine the appropriate spar and knot combination for smoothing (function can be found in r_scripts/smooth_trace.R), but is there a way to mathematically determine these values? I think this may involve some cross validation or AIC, but the cross validation code I've tried so far is very slow. 

I would also like to investigate using depth bounds in smoothing, such that there would be higher resolution of spline smoothing within a dive to retain wiggles, but lower resolution near depth = 0 to help reduce transducer noise.

**Comment by dschwilk** [comment]
Does the transducer produce more noise near zero? This seems plausible to me (eg less spring tension or something), but you would not want to change the smoothing depending on y aalues unless that was true.

**Comment by EmmaLiTsai** [comment]
Yes, the tension on the transducer arm is less at depth = 0, which causes more chatter at shallower depths. I got this information from: Kooyman, Billups & Farwell, 1983; two recently developed records for monitoring diving activity of marine birds and mammals. 

The new smoothing function I made seems to handle this decently well (smooth_trace_bounded function in r_scripts/smooth_trace.R). I'm still working on some cross validation methods to see if it would be possible to mathematically determine some of the smoothing parameters. 

I'll go ahead and assign this one to myself!
